### ISSUE TITLE
"move_and_collide" with "test_only=true" leads to a collision that should not happen

### ISSUE THREAD

**Issue by pedrohpf** [godotengine/godot#69882]
### Godot version

4.0.beta6

### System information

Windows 10

### Issue description

I was trying to code a custom function to jump.
Thus, I decided to use move_and_collide with test_only=true to check if there was a collision towards the ground.
If there was a collision, a normal of the collision was obtained to jump in the normal's direction.

It seemed to work correctly at first, but when jumping in the air near walls, a collision occurs that propels the player off the wall.
This happens even though the ground direction that is being checked is parallel to the walls and there should be no collision.
This only happens when using test_only=true.

### Steps to reproduce

Space to jump
Check console to see the usual values
Use WASD while in air to hop near a wall
When near a wall, jump and in the air press jump again
Ball will jump off the wall
Check console again to see weird values
This only happens when move_and_collide's "test_only" input is true

### Minimal reproduction project

[SimpleGameIssue.zip](https://github.com/godotengine/godot/files/10201191/SimpleGameIssue.zip)

Basically, when the bug happens:
A direction goes into move_and_collide (printed on console).
A collision is output, even though there was nothing towards the direction.
The normal output is off the nearest wall (also printed on console).

The main problem can be analyzed in "isOnGround" function.
I'm sorry if the code is confusing, I tried my best to isolate the problem and I'm pretty sure the rest can be ignored. You can just look at the input and outputs of move_and_collide and what is happening on screen.
